### PR TITLE
fix: Use npm-run-all to run the develop command

### LINKED ISSUE #34627: [Bug] npm run develop does not work on Windows machine
Another user (@cmccormack and I who both uses Windows, have run into a problem with the latest master updates when it comes to trying to running `npm run develop` on our local machines.

See the following when I tried running `npm run develop` in git-bash:
```
$ npm run develop

> @freecodecamp/freecodecamp@0.0.1 develop D:\coding\fcc
> npm-run-all -s ensure-env start-develop


> @freecodecamp/freecodecamp@0.0.1 ensure-env D:\coding\fcc
> cross-env DEBUG=fcc:* node ./tools/scripts/ensure-env.js

  fcc:tools:ensure-env ignoring creation of redirect file in undefined +0ms
  fcc:tools:ensure-env pathMigration present +11ms

> @freecodecamp/freecodecamp@0.0.1 start-develop D:\coding\fcc
> node ./tools/scripts/start-develop.js


> @freecodecamp/client@1.0.0 predevelop D:\Coding\fcc\client
> npm run prebuild

Debugger listening on ws://127.0.0.1:9229/ce7034de-7844-41f5-ab4a-bccf273b8f06
For help see https://nodejs.org/en/docs/inspector

> @freecodecamp/client@1.0.0 prebuild D:\Coding\fcc\client
> npm run build:frame-runner && node ../tools/scripts/ensure-env.js


> @freecodecamp/client@1.0.0 build:frame-runner D:\Coding\fcc\client
> webpack --env.production --config ./webpack-frame-runner.js

  fcc:server:datasources using MailHog server on port 1025 +0ms
  fcc:server:utils:about no google applications credentials environmental variable found
  fcc:server:utils:about 'GOOGLE_APPLICATION_CREDENTIALS'
  fcc:server:utils:about 'activeUser' api will always return 0
  fcc:server:utils:about this can safely be ignored during development +3s
  fcc:models:user setting up user hooks +934ms
  fcc:server freeCodeCamp server listening on port 3000 in development +4s
  fcc:server connecting to db at mongodb://localhost:27017/freecodecamp +26ms
Hash: d0144a0e981d544fd117
Version: webpack 4.22.0
Time: 10622ms
Built at: 2018-12-10 19:14:18
                Asset      Size  Chunks             Chunk Names
      frame-runner.js   235 KiB       0  [emitted]  frame-runner
  frame-runner.js.map  1.18 MiB       0  [emitted]  frame-runner
      sass-compile.js  1.13 KiB       1  [emitted]  sass-compile
  sass-compile.js.map  4.89 KiB       1  [emitted]  sass-compile
    test-evaluator.js   156 KiB       2  [emitted]  test-evaluator
test-evaluator.js.map   811 KiB       2  [emitted]  test-evaluator
Entrypoint frame-runner = frame-runner.js frame-runner.js.map
Entrypoint sass-compile = sass-compile.js sass-compile.js.map
Entrypoint test-evaluator = test-evaluator.js test-evaluator.js.map
 [92] (webpack)/buildin/global.js 489 bytes {0} {2} [built]
      ModuleConcatenation bailout: Module is not an ECMAScript module
[320] ./src/client/frame-runner.js 4.13 KiB {0} [built]
      ModuleConcatenation bailout: Module uses eval()
[321] ./src/client/workers/sass-compile.js 253 bytes {1} [built]
      ModuleConcatenation bailout: Module is not an ECMAScript module
[322] ./src/client/workers/test-evaluator.js 2.49 KiB {2} [built]
      ModuleConcatenation bailout: Module uses eval()
    + 324 hidden modules

> @freecodecamp/client@1.0.0 develop D:\Coding\fcc\client
> gatsby develop

  fcc:server db connected +1s

  fcc:boot:donate activeDonator count: 0 +1s
No Stripe API keys were found, moving on...
```
It hung on that last part.  Both @cmccormack and I waited a very long time to no avail.

I even deleted my local repo and re-cloned and still was not able to make it work.

The only way to get it to work is a two create two session windows.  In the first session you need to switch to `api-server/` and then run `node development-entry.js`.  Once it gets to the same part regarding the "No Stripe API keys were found, moving on...", in the second session, you then run `npm run develop` and it will finally work and you will see:
```
You can now view @freecodecamp/client in the browser.

  http://localhost:8000/

View GraphiQL, an in-browser IDE, to explore your site's data and schema

  http://localhost:8000/___graphql

Note that the development build is not optimized.
To create a production build, use gatsby build

i ｢wdm｣:
i ｢wdm｣: Compiled successfully.
```

@Bouncey This is the same issue you helped @cmccormack work through back on November 27th.

### PR DESCRIPTION
**Pull request by Bouncey**
## Requires Windows machine to QA correctly

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.

Closes #34627

### CODE REVIEW

**Review by cmccormack — COMMENTED** [review]
It appears to be working well - the only problem I have now is I cannot seem to kill the process(es) after the development environment is fully loaded.  I'd be interested to see if I'm the only one.

**Comment by Nirajn2311** [comment]
@cmccormack you ain’t the only. Even I couldn’t kill the processes using git bash or wsl.

**Comment by RandellDawson** [comment]
I can kill it with Ctrl+Break when using Git-Bash.

**Review by RandellDawson — COMMENTED** [review]
Works fine for me.  I can use Ctrl+Break to kill the processes when using Git-Bash

**Review by raisedadead — APPROVED** [review]

**Comment by Bouncey** [comment]
@cmccormack @Nirajn2311 I cannot work on this any longer, as I do not have a windows machine to debug on.

Could either of you take this up, or pair on it?

**Comment by RandellDawson** [comment]
@Bouncey FYI- I thought you already made changes to fix this issue a long time ago, because I no longer experience this issue on Windows 10.  I just have to make sure that I run Git-bash or Powershell as administrator and everything works fine.

**Comment by raisedadead** [comment]
> I just have to make sure that I run Git-bash or Powershell as administrator and everything works fine.

Can someone add this to the instructions in this PR? Let's get this in.

File: package.json [hunk 1]
```diff
@@ -5,14 +5,15 @@
     "postinstall": "npm run bootstrap",
     "bootstrap": "lerna bootstrap",
     "clean": "lerna clean",
-    "develop": "npm-run-all -s ensure-env start-develop",
+    "develop": "npm-run-all -s ensure-env && npm-run-all -p start-develop-server start-develop-client",
```
Comment (ValeraS) [inline package.json]:
```suggestion
    "develop": "npm run ensure-env && npm-run-all -p develop:*",
```

File: package.json [hunk 2]
```diff
@@ -5,14 +5,15 @@
...
     "ensure-env": "cross-env DEBUG=fcc:* node ./tools/scripts/ensure-env.js",
     "lint": "echo 'Warning: TODO - Define Linting with fixing.'",
     "lint:api": "eslint api-server --ignore-pattern 'api-server/node_modules/**/*.js'",
     "seed": "npm-run-all -p seed:*",
     "seed:challenges": "cross-env DEBUG=fcc:* node ./tools/scripts/seed/seedChallenges",
+    "start-develop-client": "cd ./client && node ./node_modules/gatsby-cli develop",
```
Comment (ValeraS) [inline package.json]:
```suggestion
    "develop:client": "cd ./client && npm run develop",
```

File: package.json [hunk 3]
```diff
@@ -5,14 +5,15 @@
...
+    "start-develop-server": "cd ./api-server && node development-entry.js",
```
Comment (ValeraS) [inline package.json]:
```suggestion
    "develop:server": "cd ./api-server && node development-entry.js",
```

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (raisedadead) [reply package.json]: Yep. Cleaner.

**Comment by RandellDawson** [comment]
@ValeraS Are you ok with @Bouncey's latest changes?

**Review by ValeraS — APPROVED** [review]

**Comment by RandellDawson** [comment]
> Can someone add this to the instructions in this PR? Let's get this in.

@raisedadead Can you explain what needs to be done and I will attempt to add it to this PR?

**Comment by raisedadead** [comment]
> Can you explain what needs to be done and I will attempt to add it to this PR?

Sure.

We probably need some instructions in the local setup guide. I'll leave the location in the guide itself to your discretion that adds something on the lines of:

> Note: 
> If you are on Windows, make sure the command line tool (Cmd, PowerShell or Git Bash for Windows, etc.) that you use have correct user privileges. If possible, you should launch the tool with Administrator's privilege. 
> On windows, you should be able to launch a tool by right click -> Launch as an Administrator

Feel free to make the verbiage simpler as you seem fit.

**Comment by RandellDawson** [comment]
@raisedadead OK, I added the Windows note where I think it fits the best.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**